### ISSUE TITLE
The use of regularization

### ISSUE THREAD

**Issue by ChristianEschen** [bonlime/keras-deeplab-v3-plus#12]
Thank you for the awesome code!

I observe serious overfitting on my own data using this code.
Does deeplab-v3-plus any regularization at all?

**Comment by bonlime** [comment]
This is a good point. Yes, they use l2 regularization with weight_decay=0.00004, they also use batch_norm_decay=0.9997, which is different from keras default value of 0.99.
I dont have this in my implementation, but you can add it and make pull request

**Comment by bhack** [comment]
@bonlime Probably also dropout is different. have you checked?

**Comment by bonlime** [comment]
@bhack  #4 it looks like everything is correct with dropout

**Comment by bhack** [comment]
Oh ok was already fixed.

**Comment by bonlime** [comment]
@ChristianEschen 
This model is made mainly for fine-tuning, and because of that don't use regularization as in the original paper. You can notice, that all models in Keras.applications also don't have regularization for the same reason. You can add regularization by iterating over all layers in defined model. Don't forget to compile the model after it.
`for layer in model.layers: layer.W_regularizer = l2(...)`
